xarray/backends/zarr.py

```diff
@@ -1530,9 +1532,12 @@ def open_groups_as_dict(
                     use_cftime=use_cftime,
                     decode_timedelta=decode_timedelta,
                 )
-            group_name = str(NodePath(path_group))
+            if group:
+                relative = str(NodePath("/") / NodePath(group).parent)
+                group_name = str(NodePath(path_group).relative_to(relative))
```

Comment:
If you use `parent` instead, this should work already:
```suggestion
                group_name = str(NodePath(path_group).relative_to(parent))
```

Reply:
You are right. This solved the issue